### PR TITLE
Add endpoint to retrieve top failed resources

### PR DESCRIPTION
**Pull request by GMadorell**
[sc-10229]

### CODE REVIEW

**Review by xBATx — COMMENTED** [review]
When I'm looking on this request, it sounds like it's shaped for a specific UI feature.

If you take a look on `GetResourceResultsByRuleCategoryRequest` - it contains only `passed_count` and `service_name` and `is_active` which is not here. I mean that I believe that it could be reused for this purpose too. Moreover, that request already use the composite aggregation so it should return all results.

File: com/coralogix/xdr/v1/test/security_test_service.proto [hunk 1]
```diff
@@ -193,3 +201,20 @@ message GetResourceResultsResponse {
   repeated TestResult results = 3;
   google.protobuf.Timestamp executed_at = 4;
 }
+
+message GetResourcesFailedTestCountsRequest {
+  optional google.protobuf.StringValue execution_id = 1;
+  optional com.coralogix.xdr.v1.filter.SecurityTestFilter filter = 2;
```
Comment (xBATx) [inline com/coralogix/xdr/v1/test/security_test_service.proto:207]: After @stsatlantis another PR in this repo, the `ResourceFilter` should be used here.

**Review by xBATx — APPROVED** [review]
In general looks good. I just personally think that we could reuse existing endpoint for that.

Also I think that we should wait for Barni's changes before moving this ahead to make sure the proper filter is being used.

File: com/coralogix/xdr/v1/test/security_test_service.proto [hunk 2]
```diff
@@ -193,3 +201,20 @@ message GetResourceResultsResponse {
   repeated TestResult results = 3;
   google.protobuf.Timestamp executed_at = 4;
 }
+
+message GetResourcesFailedTestCountsRequest {
+  optional google.protobuf.StringValue execution_id = 1;
```
Comment (xBATx) [inline com/coralogix/xdr/v1/test/security_test_service.proto:206]: We could add amount of top failed resources to the request params to make this endpoint more reusable in the future.

Comment (GMadorell) [reply com/coralogix/xdr/v1/test/security_test_service.proto:206]: Good point, let's add it to the request!

File: com/coralogix/xdr/v1/test/security_test_service.proto [hunk 3]
```diff
@@ -73,6 +73,14 @@ service SecurityTestService {
       body: "*"
     };
   }
+  
+  rpc GetResourcesFailedTestCounts(GetResourcesFailedTestCountsRequest) returns (GetResourcesFailedTestCountsResponse) {
```
Comment (xBATx) [inline com/coralogix/xdr/v1/test/security_test_service.proto:77]: Could we also rename this to better name like:
`GetResourcesFailedTestCounts` -> `GetTopFailedResourceTestCounts`

**Review by xBATx — APPROVED** [review]
LGTM 🎉

**Pull request merged**